### ISSUE TITLE
HomeConnect API unexpected values (SHEM78ZH5N/20)

### ISSUE THREAD

**Issue by Ganacheleague** [thoukydides/homebridge-homeconnect#198]
### Plugin Version

0.37.0

### Home Connect Appliance(s)

Bosch Dishwasher SHEM78ZH5N/20

### Log File

```shell
[12/30/2023, 6:08:44 PM] [homebridge-homeconnect] [Dishwasher] Off (disconnected)
[12/30/2023, 6:08:45 PM] [homebridge-homeconnect] Unrecognised Option in Home Connect API:
[12/30/2023, 6:08:45 PM] [homebridge-homeconnect]     Dishcare.Dishwasher.Option.SanitationUC
[12/30/2023, 6:08:45 PM] [homebridge-homeconnect] Unrecognised Option in Home Connect API:
[12/30/2023, 6:08:45 PM] [homebridge-homeconnect]     Dishcare.Dishwasher.Option.DelicateBasket
[12/30/2023, 6:08:45 PM] [homebridge-homeconnect] [Dishwasher] Connected
[12/30/2023, 6:08:45 PM] [homebridge-homeconnect] [Dishwasher] Off
[12/30/2023, 6:08:45 PM] [homebridge-homeconnect] [Dishwasher] Inactive (Inactive)
[12/30/2023, 6:08:46 PM] [homebridge-homeconnect] [Dishwasher] Finished reading available program options
[12/30/2023, 6:08:46 PM] [homebridge-homeconnect] [Dishwasher] Adding services for 6 programs
[12/30/2023, 6:08:46 PM] [homebridge-homeconnect] [Dishwasher]     Heavy     (Dishcare.Dishwasher.Program.Intensiv70)  
[12/30/2023, 6:08:46 PM] [homebridge-homeconnect] [Dishwasher]     Auto      (Dishcare.Dishwasher.Program.Auto2)       
[12/30/2023, 6:08:46 PM] [homebridge-homeconnect] [Dishwasher]     Delicate  (Dishcare.Dishwasher.Program.Glas40)      
[12/30/2023, 6:08:46 PM] [homebridge-homeconnect] [Dishwasher]     Normal    (Dishcare.Dishwasher.Program.Eco50)       
[12/30/2023, 6:08:46 PM] [homebridge-homeconnect] [Dishwasher]     Speed 60  (Dishcare.Dishwasher.Program.Quick65)     
[12/30/2023, 6:08:46 PM] [homebridge-homeconnect] [Dishwasher]     Rinse     (Dishcare.Dishwasher.Program.PreRinse)    
[12/30/2023, 6:08:46 PM] [LGThinQ] Successfully connected to the MQTT server.
[12/30/2023, 6:08:46 PM] [homebridge-homeconnect] [Dishwasher] All features successfully initialised in 4 seconds 814 milliseconds
[12/30/2023, 6:08:46 PM] [homebridge-homeconnect] [Dishwasher] 2 optional features enabled by default (unconfigured):
[12/30/2023, 6:08:46 PM] [homebridge-homeconnect] [Dishwasher]     Event Buttons         (Stateless Programmable Switch service)
[12/30/2023, 6:08:46 PM] [homebridge-homeconnect] [Dishwasher]     Door           Doors  (Door service)
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect] Home Connect API returned keys/values that are unrecognised by this plugin
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect] Please report these by creating a new GitHub issue using this link:
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect]     https://github.com/thoukydides/homebridge-homeconnect/issues/new?template=key_value.yml&title=HomeConnect+API+unexpected+values+%28SHEM78ZH5N%2F20%29&version=0.37.0&appliance=Bosch+Dishwasher+SHEM78ZH5N%2F20
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect] Most of the issue fields will be filled-in appropriately.
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect] Just paste the following into the "Log File" field and submit the issue:
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect] â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼â¼
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect] 
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect] // Option
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect] export interface OptionValues {
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect]     'BSH.Common.Option.EnergyForecast'?:           number;
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect]     'BSH.Common.Option.RemainingProgramTime'?:     number;
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect]     'BSH.Common.Option.StartInRelative'?:          number;
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect]     'BSH.Common.Option.WaterForecast'?:            number;
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect]     'Dishcare.Dishwasher.Option.DelicateBasket'?:  unknown; // (unrecognised)
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect]     'Dishcare.Dishwasher.Option.HalfLoad'?:        boolean;
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect]     'Dishcare.Dishwasher.Option.SanitationUC'?:    unknown; // (unrecognised)
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect]     'Dishcare.Dishwasher.Option.ZeoliteDry'?:      boolean;
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect] }
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect] 
[12/30/2023, 6:10:46 PM] [homebridge-homeconnect]
```

**Comment by thoukydides** [comment]
Thanks for the report.

Neither of those options are listed in the [Home Connect API documentation](https://api-docs.home-connect.com/programs-and-options/?#dishwasher). I will include them in the next plugin release.

**Comment by thoukydides** [comment]
Both options should be addressed in [v0.37.1](https://github.com/thoukydides/homebridge-homeconnect/releases/tag/v0.37.1).

Thanks again for the report.
